client/components/videos-ui/index.jsx

```diff
@@ -18,6 +18,17 @@ const VideosUi = ( { shouldDisplayTopLinks = false, onBackClick = () => {} } ) =
 	const [ isPlaying, setIsPlaying ] = useState( false );
 	const videoRef = useRef( null );
 
+	const [ isMobile, setIsMobile ] = useState( window.innerWidth < 600 );
+	useEffect( () => {
+		window.addEventListener(
+			'resize',
+			() => {
+				setIsMobile( window.innerWidth < 600 );
+			},
+			false
+		);
+	}, [ isMobile ] );
```

Comment:
This adds event listeners when isMobile changes, but never removes them. I recommend changing the `useEffect` to return a 'cleanup callback' which removes the event listener. This will also require extracting the second parameter from a lambda to a stable variable that can be used in both the addEventListener and removeEventListener steps. Something like:

```js
const changeme = useCallback( () => setIsMobile( window.innerWidth < 600 ) );
useEffect( () => {
  window.addEventListener( 'resize', changeme, false );
  return () => {
    window.removeEventListener( 'resize', changeme );
  };
}, [ isMobile ] );
```

Other questions:
* Does the event listener need to be reattached whenever isMobile changes? Is it okay to only set it up once (dependency array = `[]` or `[setIsMobile]`)?
* `'resize'` events can be spammy and fire several times while I make one drag adjusting my window. Also recommend checking if this is the case (console.log inside the callback to see if it's firing a lot) and potentially debouncing if this is a performance issue.

Reply:
Yes, these sorts of listeners do tend to be very noisy and maybe not worth it. I'll be trying to simplify this component in general, hopefully ending up in a place where we don't need these listeners at all. 🤔

Reply:
I actually just remembered that there are already some custom hooks set up for this. `useMobileBreakpoint()`, `useBreakpoint()`, `useDesktopBreakpoint()`.

Examples:
https://github.com/Automattic/wp-calypso/blob/427fb827098f09535bbc81198bb3eb0f1596f988/client/components/jetpack/threat-history-list/index.tsx#L61
https://github.com/Automattic/wp-calypso/blob/427fb827098f09535bbc81198bb3eb0f1596f988/client/layout/index.jsx#L51

I can't find my commit using this but I kind of remember some restrictions being there. I think it was if you're using `useBreakpoint()`, you can't use any pixel value you want, you have to use [one of these](https://github.com/Automattic/wp-calypso/blob/427fb827098f09535bbc81198bb3eb0f1596f988/packages/viewport/src/index.js#L73-L91).

Reply:
We could also use `useViewportMatch` from `@wordpress/compose` and it might be better to use `@wordpress` functionality in long term
```
import { useViewportMatch } from '@wordpress/compose';
```

Reply:
We're not going to end up needing resize event listeners for the current refactor. 👍